### PR TITLE
docs: add Centos/RHEL9+, thanks for nothing RedHat

### PR DESCRIPTION
**Pull request by janl**
we can merge this after we publish the rocky9 build rpm

### CODE REVIEW

**Comment by nickva** [comment]
Due to the generous support from IBM's Z open source support team (@kun-lu20) in https://github.com/apache/couchdb-ci/pull/54 we have RHEL9 (clone) support with the caveat that it depends on the EPEL repo.

I just uploaded the x86-64 3.3.2 centos-9 rpm to our artifactory

```
yum install couchdb
Last metadata expiration check: 0:01:36 ago on Fri Jul 21 16:00:37 2023.
Dependencies resolved.
=============================================================================================================================================================================================
 Package                                     Architecture                               Version                                            Repository                                   Size
=============================================================================================================================================================================================
Installing:
 couchdb                                     x86_64                                     3.3.2-1.el9                                        couchdb                                      36 M

Transaction Summary
=============================================================================================================================================================================================
Install  1 Package

Total download size: 36 M
Installed size: 87 M
Is this ok [y/N]: y
Downloading Packages:
couchdb-3.3.2-1.el9.x86_64.rpm                                                                                                                               4.7 MB/s |  36 MB     00:07    
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                        4.7 MB/s |  36 MB     00:07     
couchdb                                                                                                                                                       50 kB/s | 3.8 kB     00:00    
Importing GPG key 0x7A00258D:
 Userid     : "The Apache Software Foundation (Package repository signing key) <root@apache.org>"
 Fingerprint: 390E F70B B1EA 12B2 7739 6295 0EE6 2FB3 7A00 258D
 From       : https://couchdb.apache.org/repo/keys.asc
Is this ok [y/N]: y
Key imported successfully
couchdb                                                                                                                                                       32 kB/s | 3.9 kB     00:00    
Importing GPG key 0x9657A78E:
 Userid     : "Nicolae Vatamaniuc <vatamane@apache.org>"
 Fingerprint: 0BD7 A984 99C4 AB41 C910 EE65 FC04 DFBC 9657 A78E
 From       : https://couchdb.apache.org/repo/rpm-package-key.asc
Is this ok [y/N]: y
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                     1/1 
  Running scriptlet: couchdb-3.3.2-1.el9.x86_64                                                                                                                                          1/1 
  Installing       : couchdb-3.3.2-1.el9.x86_64                                                                                                                                          1/1 
  Running scriptlet: couchdb-3.3.2-1.el9.x86_64                                                                                                                                          1/1 
Generating random cookie value.

  Verifying        : couchdb-3.3.2-1.el9.x86_64                                                                                                                                          1/1 

Installed:
  couchdb-3.3.2-1.el9.x86_64                                                                                                                                                                 

Complete!
```

File: src/docs/src/install/unix.rst [hunk 1]
```diff
@@ -63,11 +64,19 @@ Enabling the Apache CouchDB package repository
...
     sudo yum install -y yum-utils
     sudo yum-config-manager --add-repo https://couchdb.apache.org/repo/couchdb.repo
 
+**RedHat(>=9)**: Run the following commands::
+
+    sudo yum install -y yum-utils
+    sudo yum-config-manager --add-repo https://couchdb.apache.org/repo/couchdb.repo
+    # Enable EPEL for the SpiderMonkey dependency
+    sudo dnf config-manager --set-enabled crb
+    sudo dnf install epel-release epel-next-release
```
Comment (nickva) [inline src/docs/src/install/unix.rst:78]: For me on Almalinux 9 VM adding `epel-next-release` wasn't needed. I used just `epel-release`

**Review by nickva — APPROVED** [review]
+1

Package created, signed and was uploaded. See a minor comment about epel-next. I tried it on AlmaLinux 9 not sure if it's a Rocky vs Centos9 vs Alma thing or something else.

> thanks for nothing RedHat

Heh ;-) Tbh, I think this is just as much a "Thanks for nothing, Mozilla". I can't blame Redhat not wanting to support Spidermonkey 78 for [15 years](https://bugzilla.redhat.com/show_bug.cgi?id=1958111 ). I wouldn't either.

We have the same problem as Redhat as SM keeps changing their "stable" API with every release (see https://github.com/apache/couchdb/pull/4305) That stuff works for a browser, but not tenable for an "embedded" engine.

**Comment by janl** [comment]
> "Thanks for nothing, Mozilla"

totes fair :) This is just happening at the same time as RedHat is trying to take the RHEL sources private, which creates uncertainty and potentially more work for us for no good reason :)

File: src/docs/src/install/unix.rst (hunk 1 above)
Comment (janl) [reply src/docs/src/install/unix.rst:78]: Good catch! That’s why I specifically pulled out RedHat >=9 from RedHat/CentOS <=9) — I hope that helps. IF we get reports that the “next” bit is wrong, we can adjust on the fly.

Comment (nickva) [reply src/docs/src/install/unix.rst:78]: Ah good point. It's a RHEL 9 only instructions. Most people would probably figure the Almalinux / Rocky 9 instructions. If we got a lot questions we can add them later.

**Pull request merged**